e2e-tests/strategies.sh

```diff
@@ -49,3 +49,11 @@ if [ "$exit_code" != 94 ]; then
     echo "Expected exit code 94, but actual exit code $exit_code"
     exit 1
 fi
+
+set -euxo pipefail
+
+## Test --strategies overriding `disabled-strategies=["compile"]` in Cargo.toml
+ "./$1" binstall --no-confirm --manifest-path "manifests/strategies-test-override-Cargo.toml" --strategies compile cargo-quickinstall@0.2.10
+
+## Test --disable-strategies overriding `disabled-strategies=["compile"]` in Cargo.toml
```

Comment:
I only glanced at this so far, but my first instinct would be to expect the CLI --disable-strategies to only ever *decrease* the set of allowed strategies. So, I would make it so the situation of this test disallow all strategies.

Reply:
I think the current implementation might makes sense?

After all cli options can overrides anything from the maintainer, previously we have pkg-url, pkg-bin and pkg-fmt which overrides the maintainers' configurations.

Reply:
I think `disabled-strategies` should be treated more like specifying both on the same command-line.

Here's a scenario: let's say the maintainer disabled `quick-install`, but the user doesn't know about it. Let's say Github downloads are temporarily down for whatever reason. The user runs `cargo binstall package`, it uses `compile`, and they don't like it. So, one natural thing to try would be to interrupt that and try `cargo binstall --disable-strategies compile`.

I think that it wouldn't be great if this installed the package from quickinstall without the user realizing that he's getting a different thing than the GitHub download they usually gets. I think the "good" outcome here would be if `binstall` told the user that it can't download the package from GitHub, and thus can't install anything. (It could also tell the user that quick-install was disabled by maintainer, and to use `--strategies quick-install` if the user wants to override that).

I don't think this scenario would be very common, but it's enough to form my opinion of what behavior is a bit better. Either way, we should make sure that what happens is documented.

Reply:
Yeah, maybe we should add a new option for this...

Reply:
cc @passcod I don't have much idea on how we could fix/improve this, do you have any idea/suggestion for this?